### DISCUSSION TITLE
Forked repos with Forking Renovate

### DISCUSSION THREAD

**Discussion by rarkins** [renovatebot/renovate#21057] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

I discovered that there's a limitation when it comes to forked repositories and Renovate's fork mode.

Let's say that a repo and all of it's forks and descendants are referred to as a "family" here.

GitHub only allows one family member per account/org.

So if we have an account/org we fork into, GitHub only allows there to be one family fork. This means a conflict or problem if either:
- A repo and one of its forks or descendants are both installed, or
- Two forks of the same ancestor are both installed

One simple way to avoid this would be to disable forks when in fork mode. I looked in the Forking Renovate app and here's a list of installed forked repos which were part of "Select repos" instead of "All repos":

<img width="384" alt="image" src="https://user-images.githubusercontent.com/6311784/226523157-ca1135e2-56bf-4512-9b49-8f65831092bd.png">

(the above are all public, so not giving away any confidential info)

@Chumper do you recall why you installed kingpin?

@devlauer you appear to have an active fork in https://github.com/devlauer/testcontainers-examples

@grzesuav you also have one, maybe temporarily for testing?

It seems like:
- We need to block forks by default, when in fork mode
- We might have the option for administrators to configure them on if they're really sure

**Comment by codeboten** [comment]
@rarkins I think I ran into this issue when testing out forking-renovate for forks of some repository (open-telemetry/opentelemetry-collector & open-telemetry/opentelemetry-collector-contrib) in the OpenTelemetry project. The forks of those repos were created from my personal fork ending up https://github.com/renovate-bot/codeboten-_-opentelemetry-collector and https://github.com/renovate-bot/codeboten-_-opentelemetry-collector-contrib

What's the correct way to move forward from this state?

**Reply by rarkins** [reply]
Forking renovate can't support forked repos due to the fundamental restrictions GitHub has. You need to self host or use the regular Renovate

**Reply by codeboten** [reply]
Right, so I tested things in the fork, but once I validated forking renovate could do what we needed, I want to use it for the original repo. My question is it possible to fix the state that currently exists? in other words, am I stuck w/ the forks that renovatebot created (with the `codeboten-_-` prefix) or is it possible to remove those forks from the renovate-bot repo so that `open-telemetry/opentelemetry-collector` and `open-telemetry/opentelemetry-collector-contrib` re-create clean repos under `renovate-bot` 

Here's an example PR that was created against `open-telemetry/opentelemetry-collector` by renovate-bot from the fork of the fork https://github.com/open-telemetry/opentelemetry-collector/pull/7722. I would like this PR to be from a fork of the original repo, rather than the fork of the fork :)

**Reply by rarkins** [reply]
I think we can manually clean it up by deleting the fork's fork but probably any open PRs get closed by GitHub when we do that

**Reply by codeboten** [reply]
That would be great. Would it be helpful for me to create an issue for this? I didn't mean to hijack this discussion topic

**Reply by rarkins** [reply]
I've made the two deletions. If any further problems, please create a new Discussion (not Issue)

**Reply by codeboten** [reply]
Thanks for the quick turnaround! I'm not sure but I'm still seeing this repo here: https://github.com/renovate-bot/codeboten-_-opentelemetry-collector, can you confirm it was deleted?

**Reply by rarkins** [reply]
I can confirm I 100% deleted it just now. I don't remember for sure if it was one of the ones I removed earlier, so let's see if it stays gone. You've uninstalled the app personally, right?

**Reply by codeboten** [reply]
I've uninstalled it in my personal repo 👍🏻 

Once again, much appreciate the help!
